File: tests/benchmark/corpus/java/interprocedural/InterprocTaintPropagation.java
import javax.servlet.http.*;

public class InterprocTaintPropagation extends HttpServlet {
    private String buildQuery(String userFilter) {
        return "SELECT * FROM logs WHERE msg LIKE '%" + userFilter + "%'";
    }

    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Exception {
        String filter = req.getParameter("filter");
        String query = buildQuery(filter);
        java.sql.Connection conn = null;
        conn.createStatement().executeQuery(query);
    }
}

File: tests/benchmark/corpus/go/interprocedural/interproc_taint_propagation.go
package main

import (
	"database/sql"
	"net/http"
)

func buildQuery(filter string) string {
	return "SELECT * FROM logs WHERE msg LIKE '%" + filter + "%'"
}

func handler(w http.ResponseWriter, r *http.Request) {
	filter := r.URL.Query().Get("filter")
	query := buildQuery(filter)
	var db *sql.DB
	db.Query(query)
}
